airbyte-cdk/python/airbyte_cdk/destinations/destination.py

```diff
@@ -30,37 +30,39 @@
 
 from airbyte_cdk import AirbyteLogger
 from airbyte_cdk.connector import Connector
-from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog, Type
+from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog, ConnectorSpecification, Type
+from airbyte_cdk.sources.utils.schema_helpers import check_config_against_spec
 from pydantic import ValidationError
 
 
 class Destination(Connector, ABC):
     logger = AirbyteLogger()
+    VALID_CMDS = {"spec", "check", "write"}
 
     @abstractmethod
     def write(
         self, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog, input_messages: Iterable[AirbyteMessage]
     ) -> Iterable[AirbyteMessage]:
         """Implement to define how the connector writes data to the destination"""
 
-    def _run_spec(self) -> AirbyteMessage:
-        return AirbyteMessage(type=Type.SPEC, spec=self.spec(self.logger))
+    def _run_spec(self, spec: ConnectorSpecification) -> AirbyteMessage:
```

Comment:
we should delete this method, calling it `_run_spec` when it just wraps an object is misleading

Reply:
Fixed